src/memcache/sockets.cpp

```diff
@@ -101,7 +89,7 @@ memcache_socket::memcache_socket(int fd,
                                           << len << " bytes.";
                 buf.reset();
                 unlock_all();
-                invalidate_and_close();
+                with_fd([](int) -> bool { return false; });
```

Comment:
This usage appears many times, but it is difficult to understand the intent at first sight.
It may be better to create a dedicated method.

Reply:
will do.